mlflow/models/evaluation/default_evaluator.py

```diff
@@ -784,11 +784,13 @@ def _log_and_return_evaluation_result(self):
 
     def _evaluate_classifier(self):
         from mlflow.models.evaluation.lift_curve import plot_lift_curve
+        from mlflow.sklearn import _AUTOLOGGING_METRICS_MANAGER
 
         self.label_list = np.unique(self.y)
         self.num_classes = len(self.label_list)
 
-        self.y_pred = self.predict_fn(self.X)
+        with _AUTOLOGGING_METRICS_MANAGER.disable_log_post_training_metrics():
+            self.y_pred = self.predict_fn(self.X)
```

Comment:
The predict function is also used in shap explainer, we should also disable it.

I think we can put `_AUTOLOGGING_METRICS_MANAGER.disable_log_post_training_metrics()` manager in the outer scope

Reply:
Good idea, fixed.